dpctl/tensor/_permute_dims.py

```diff
@@ -0,0 +1,52 @@
+#                       Data Parallel Control (dpctl)
+#
+#  Copyright 2020-2021 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+from itertools import permutations
+
+import dpctl.tensor as dpt
+
+
+def permute_dims(X, axes):
+    """
+    permute_dims(X: usm_ndarray, axes: tuple or list) -> usm_ndarray
+
+    Permutes the axes(dimensions) of the given usm_ndarray. Returns a view
+    containing the axes permutation.
+    """
+    if not isinstance(X, dpt.usm_ndarray):
+        raise TypeError(f"Expected usm_ndarray type, got {type(X)}.")
+    if not isinstance(axes, (tuple, list)):
+        axes = (axes,)
+    if not X.ndim == len(axes):
+        raise ValueError(
+            "The length of the passed axes does not match "
+            "to the number of usm_ndarray dimensions."
+        )
+    if axes not in permutations(range(0, X.ndim), X.ndim):
```

Comment:
This has horrible computational and space complexity in the dimensionality of the array `d`, specifically `O(d!)`. 

More efficient way:
   - Length of the tuple must be `d`
   - Minimum element must be `0`, maximum `d-1`.
   - Array should have no duplicates.